### ISSUE TITLE
only book doctypes can contain level 0 sections

### ISSUE THREAD

**Issue by robogeek** [asciidoctor/asciidoctor.js#358]
I'm looking to integrate AsciiDoc into AkashaCMS.  I've created a sample file to test things, the sample is built using snippets from the documentation on asciidoctor.org

Rendering occurs this way:

```js
const asciidoctor = require('asciidoctor.js')();


const options = {
    doctype: 'inline',
    attributes: 'showtitle'
};
```

Then 

```js
    renderSync(text, metadata) {
        // console.log('AsciidocRenderer renderSync '+ text);
        var ret = asciidoctor.convert(text, options);
        // console.log(ret);
        return ret;
    }

    render(text, metadata) {
        // console.log('AsciidocRenderer render');
        return new Promise((resolve, reject) => {
            try {
                resolve(asciidoctor.convert(text, options));
            } catch(e) {
                reject(e);
            }
        });
    }
```

I get an error: `asciidoctor: ERROR: <stdin>: line 4: only book doctypes can contain level 0 sections`

That line corresponds to the Document Title in this:

```adoc
See http://asciidoctor.org for documentation

Document Title
==============
Doc Writer <thedoc@asciidoctor.org>
:idprefix: id_

Preamble paragraph.

NOTE: This is test, only a test.

== Section A

*Section A* paragraph.

=== Section A Subsection

*Section A* 'subsection' paragraph.

== Section B

*Section B* paragraph.

.Section B list
* Item 1
* Item 2
* Item 3
```

What I found after some searching is that there's a "book" doctype that can be set which allows Level 0 Sections.  But your documentation doesn't describe any way to set options like `doctype`.

The `options` object shown above came from perusing Asciidoctor support from some other frameworks.

If I change that object to:

```js
const options = {
    doctype: 'book',
    attributes: 'showtitle'
};
```

Then the error message goes away.

Bottom line is this `options` argument is not defined nor discussed and the source code is not easy to follow.

**Comment by mojavelinux** [comment]
You can find information about the available doctypes and what they do here:

* http://asciidoctor.org/docs/user-manual/#document-types
* http://asciidoctor.org/docs/user-manual/#sections

The inline doctype is only for converting a single phrase or paragraph. Most likely you'll want to use article (default if not specified) or book.

If you only want embeddable HTML, then you set the header_footer option to false (which is the default when using the API).

**Comment by mojavelinux** [comment]
```
See http://asciidoctor.org for documentation

Document Title
==============
Doc Writer <thedoc@asciidoctor.org>
:idprefix: id_
```

You are attempting to use content above the document title. That is not permitted. This is described here:

* http://asciidoctor.org/docs/user-manual/#doc-header

**Comment by mojavelinux** [comment]
> I'm looking to integrate AsciiDoc into AkashaCMS.

I'm excited to here how it goes.

**Comment by robogeek** [comment]
I'm new to Asciidoc and know hardly anything.  Shouldn't the options object be documented in your documentation as well?

**Comment by mojavelinux** [comment]
Eventually, all the documentation will be aggregated at asciidoctor.org. We grew quickly and asciidoctor.org hasn't had a chance to catch up. But it's something we'll be working on. To avoid documenting options in multiple places (which will diverge), part of that is coming up with a way to provide a reusable description of the options.

**Comment by Mogztter** [comment]
Documentation is happening... https://asciidoctor-docs.netlify.com/asciidoctor.js/processor/convert-options/

**Comment by mklueh** [comment]
Why is there even such a restriction?

**Comment by mojavelinux** [comment]
Because it's how the processor knows how to process the document...and more specifically how to create valid DocBook for that backend. This is not something that is going to be changed unless it gets raised in the AsciiDoc Language project.
